src/main/java/cloud/fogbow/common/plugins/cloudidp/cloudstack/CloudStackIdentityProviderPlugin.java

```diff
@@ -43,13 +48,20 @@ public CloudStackUser getCloudUser(Map<String, String> credentials) throws Fogbo
         // send a valid json body in the post request
         HttpResponse response = HttpRequestClient.doGenericRequest(HttpMethod.POST,
                 request.getUriBuilder().toString(), new HashMap<>(), new HashMap<>());
+        // NOTE(pauloewerton): we need to extract all set-cookie headers in order to pass it to the follow-on requests
+        Map<String, List<String>> headerFields = response.getHeaders();
+        String setCookieHeaders1 = String.join(COOKIE_SEPARATOR, headerFields.get(SET_COOKIE_HEADER_1));
+        String setCookieHeaders2 = String.join(COOKIE_SEPARATOR, headerFields.get(SET_COOKIE_HEADER_2));
+        String cookieHeaderValue = String.join(COOKIE_SEPARATOR, setCookieHeaders1 , setCookieHeaders2);
+        HashMap<String, String> cookieHeader = new HashMap<>();
+        cookieHeader.put(COOKIE_HEADER, cookieHeaderValue);
```

Comment:
Maybe lines 51 through 57 could be extracted into a private method called 'getCookieHeaders' for improved readability.

Reply:
sure